modules/images/webp-uploads/load.php

```diff
@@ -684,6 +684,135 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
+		$image_file = $main_images[ $targeted_mime ]['path'];
+
+		if ( ! file_exists( $image_file ) ) {
+			continue;
+		}
+
+		$metadata['sources'][ $targeted_mime ] = array(
+			'file'     => wp_basename( $image_file ),
+			'filesize' => filesize( $image_file ),
+		);
+
+		foreach ( $metadata['sizes'] as $size => $size_details ) {
+			// Add sources to resized image metadata.
+			$image_file = $original_directory . '/' . $subsized_images[ $targeted_mime ][ $size ]['file'];
+
+			if ( ! file_exists( $image_file ) ) {
+				continue;
+			}
+
+			$metadata['sizes'][ $size ]['sources'][ $targeted_mime ] = array(
+				'file'     => wp_basename( $image_file ),
+				'filesize' => filesize( $image_file ),
+			);
+		}
+	}
+
+	return $metadata;
+}
+
+/**
+ * Creates additional image formats when original image is edited.
+ *
+ * @since n.e.x.t
+ *
+ * @param bool|null       $override  Value to return instead of saving. Default null.
+ * @param string          $file      Name of the file to be saved.
+ * @param WP_Image_Editor $image     The image editor instance.
+ * @param string          $mime_type The mime type of the image.
+ * @param int             $post_id   Attachment post ID.
+ *
+ * @return bool|null
+ */
+function webp_uploads_update_image_onchange( $override, $file, $image, $mime_type, $post_id ) {
+	if ( null !== $override ) {
+		return $override;
+	}
+
+	if ( empty( webp_uploads_get_upload_image_mime_transforms()[ $mime_type ] ) ) {
+		return null;
+	}
+
+	add_filter(
+		'wp_update_attachment_metadata',
+		function ( $metadata, $post_meta_id ) use ( $post_id, $file, $mime_type, $image ) {
+			if ( $post_meta_id !== $post_id ) {
+				return $metadata;
+			}
+
+			$valid_mime_transforms = webp_uploads_get_upload_image_mime_transforms()[ $mime_type ];
+			$original_directory    = pathinfo( $file, PATHINFO_DIRNAME );
+			$filename              = pathinfo( $file, PATHINFO_FILENAME );
+			$allowed_mimes         = array_flip( wp_get_mime_types() );
+			$old_metadata          = wp_get_attachment_metadata( $post_id );
+			$resize_sizes          = array();
+			$main_images           = array();
+			$subsized_images       = array();
+
+			foreach ( $old_metadata['sizes'] as $size_name => $size_details ) {
+				if ( isset( $metadata['sizes'][ $size_name ] ) && ! empty( $metadata['sizes'][ $size_name ] ) &&
+					$metadata['sizes'][ $size_name ]['file'] !== $old_metadata['sizes'][ $size_name ]['file'] ) {
+					$resize_sizes[ $size_name ] = $metadata['sizes'][ $size_name ];
+				}
+			}
+
+			foreach ( $valid_mime_transforms as $targeted_mime ) {
+				if ( $targeted_mime === $mime_type ) {
+					$main_images[ $targeted_mime ]     = array( 'path' => $file );
+					$subsized_images[ $targeted_mime ] = $metadata['sizes'];
+					continue;
+				}
+
+				if ( ! isset( $allowed_mimes[ $targeted_mime ] ) || ! is_string( $allowed_mimes[ $targeted_mime ] ) ) {
+					return null;
+				}
+
+				if ( ! wp_image_editor_supports( array( 'mime_type' => $targeted_mime ) ) ) {
```

Comment:
Technically, this check is not fully reliable, since it checks if _any_ of the available image editors supports the MIME type. But we're already using a specific image editor here (within the `$image` variable), so we need to check if _that_ image editor supports the target MIME type.

You can see how we can do that by checking the internals of `wp_image_editor_supports()` and `_wp_image_editor_choose()` - basically we have to rely on (whether there is) a `supports_mime_type()` method on `$image`.